src/mode.js

```diff
@@ -296,6 +297,17 @@ function type(style) {
   };
 }
 
+function constraints(rule, exclusions) {
+  rule.constraints = token => {
+    for (let ex of exclusions) {
+      if (token.value.match(ex)) return true;
+    }
+    return false;
+  };
+
+  return rule;
+}
+
```

Comment:
I'd suggest moving this right up under `opt` and calling it `butNot`

Also, I think you can define this more cleanly with a `some`

```
function butNot(rule, exclusions) {
  return token => exclusions.some(exclusion => token.value.match(exclusion));
}